### ISSUE TITLE
latent share

### ISSUE THREAD

**Issue by msankith** [sdc17/SwiReasoning#3]
Hi authors—thanks for releasing SwiReasoning!

**What I observed**

I’m consistently seeing very low latent usage. With window_size from 32 → 512, the latent-phase proportion ranges ~0.4% → 3.5% of total generation steps.

As window_size decreases, latent usage increases slightly, but accuracy drops.

At window_size=512, accuracy is good; however, the time reduction mostly comes from the Convergence/Termination trigger logic, not from latent segments. In my runs, latent accounts for only ~0.5% of tokens/steps at 512.

Model: Qwen3-8B
Dataset: GSM-8K

**Why I’m confused**

The paper shows a window-size ablation where 512 gives the best average accuracy (Table 3), which I also see. But I don’t find any report of the share of latent vs explicit steps or time attribution between (i) latent blocks vs (ii) convergence/early-answer triggers vs (iii) termination triggers.

Algorithm 1 (Appendix) explicitly inserts a “Convergence trigger” when in explicit mode and within the switch-count bounds—this looks like a strong source of speedup in practice.

**Questions**

1. Is very low latent usage expected? With your default hyper-params (α/β schedule, WE→L window, C_max), what latent share do you typically see on GSM8K/MATH/GPQA/AIME?

2. How much of the reported speedup is from latent vs triggers? Any breakdown of (a) time/tokens in latent, (b) time saved by convergence/early-answer checkpoints, and (c) termination behavior?

**Comment by sdc17** [comment]
Hi @msankith,

Thank you for your interest in our work and verifying the accuracy! 

Q1. The observation of low latent share is largely due to and expected with the asymmetric switch design introduced in Sec. 3.3. In early explorations we tested a symmetric version and found it could hurt final accuracy from time to time. You could try the symmetric version offering high latent share, which raises the latent share to 40%+ by simply adding an `allow_switch` flag into the line

https://github.com/sdc17/SwiReasoning/blob/5afe509bc3c2af3f33cb65beb317280d30319d1f/generation_utils.py#L258

That is to say, modifying it to be

```python
 to_normal = (mode == 0) & (cur_entropy < cur_ref_entropy) & allow_switch
```
We'd suggest a much smaller `window_size` like around 8 for this version. The trade-off is that accuracy can drop on some tasks when latent segments become too long/frequent. Results will vary with the model/dataset. Feel free to play with it.

Q2. The token saving of applying vanilla latent reasoning is modest, specifically for a training-free setting. For example, in Soft-Thinking paper, by directly applying latent reasoning, the token saving is 10%-20% on average. The proposed convergence and termination triggers are mechanisms complementing SwiReasoning to achieve more significant token savings by answering earlier when partial reasoning trajectories already suffices for a correct answer.

Thanks.
